api/predict.py

```diff
@@ -63,6 +63,11 @@ def post(self):
 
         # Aligning the predictions to the required API format
         label_preds = [{'label_id': p[0], 'label': p[1], 'probability': p[2]} for p in preds]
+
+        # Filter list
+        if args['filter'] is not None and args['filter'] != ['']:
```

Comment:
Do you think it's better to check all entries in `args['filter']`?

```python
if args['filter'] is not None and all(x.strip() != '' for x in args['filter']):
```

Reply:
Hmmm. I'm not sure if it's necessary. I added the 2nd part of the test to catch instances where filter was passed but left blank (ex: `http://localhost:5000/model/predict?filter=`).

If somebody passed filter arguments but one of them was blank (ex `http://localhost:5000/model/predict?filter=,,Clang,Ding`) we should still filter based on the known arguments.

Reply:
I agree with you that known arguments should be used, but it looks like the current code would break if more than one blank entry is given:

    http://localhost:5000/model/predict?filter=,,

Reply:
I see your point. I will fix this and add a test.